common/app/views/support/cleaner/AmpEmbedCleaner.scala

```diff
@@ -42,28 +42,35 @@ case class AmpEmbedCleaner(article: Article) extends HtmlCleaner {
     })
   }
 
-  sealed abstract class AmpExternalVideo(val videoId: String, val elementType: String)
-  case class YoutubeExternalVideo(override val videoId: String) extends AmpExternalVideo(videoId, "amp-youtube")
-  case class VimeoExternalVideo(override val videoId: String) extends AmpExternalVideo(videoId, "amp-vimeo")
 
+  sealed abstract class AmpExternalVideo(val videoId: String, val elementType: String, customAttributes: String => Map[String, String]){
+    def getAttributes = customAttributes(videoId) ++ Map(("width", "5"), ("height", "3"), ("layout", "responsive"))
+  }
+
+  case class YoutubeExternalVideo(override val videoId: String) extends AmpExternalVideo(videoId, "amp-youtube", customAttributes = id => Map(("data-videoid", id)))
+  case class VimeoExternalVideo(override val videoId: String) extends AmpExternalVideo(videoId, "amp-vimeo", customAttributes = id => Map(("data-videoid", id)))
+  case class FacebookExternalVideo(override val videoId: String) extends AmpExternalVideo(videoId, "amp-facebook", customAttributes = id => Map(("data-href", s"https://www.facebook.com/theguardian/videos/$id"), ("data-embed-as", "video")))
 
+  
   object AmpExternalVideo {
```

Comment:
General thought about this file not really related to these changes: it would be nice to put `private` on everything that's intended to be.